core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -121,180 +121,148 @@ public TableOperations underlyingOps() {
     return ops;
   }
 
-  private void checkLastOperationCommitted(String operation) {
+  protected <T extends PendingUpdate> T appendUpdates(T update) {
```

Comment:
This is only appending a single update so it should not be plural. How about `appendUpdate` instead?